### ISSUE TITLE
Quarkus 2.7 breaks Flyway with MariaDB.

### ISSUE THREAD

**Issue by michael-simons** [quarkusio/quarkus#23474]
### Describe the bug

When using the official Quarkus Flyway extension together with Quarkus JDBC Maria DB, tests using devservices will fail with:

```
Caused by: org.flywaydb.core.api.FlywayException: Unsupported Database: MariaDB 10.5
	at org.flywaydb.core.internal.database.DatabaseTypeRegister.getDatabaseTypeForConnection(DatabaseTypeRegister.java:106)
```

That used to worked in 2.6.x.

### Expected behavior

Migrations should be applied.

### Actual behavior

Flyway fails, see above.

### How to Reproduce?

Have some migration and configure migrate on startup

```
quarkus.datasource.db-kind=mariadb

quarkus.flyway.migrate-at-start=true
```

required dependencies

```xml
<dependency>
			<groupId>io.quarkus</groupId>
			<artifactId>quarkus-flyway</artifactId>
		</dependency>
		<dependency>
			<groupId>io.quarkus</groupId>
			<artifactId>quarkus-jdbc-mariadb</artifactId>
		</dependency>
```

*Reproducer is attached: [demo.zip](https://github.com/quarkusio/quarkus/files/8014150/demo.zip)*

### Output of `java -version`

java version "11.0.7" 2020-04-14 LTS

### Quarkus version or git rev

2.7.0

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Maven

### Additional information

Another reproducer (where i discovered the actual issue)

https://github.com/michael-simons/scrobbles4j/runs/5090741197?check_suite_focus=true#step:5:704

**Comment by geoand** [comment]
This actually isn't a Quarkus issue, it's due to Flyway moving MySQL code to a different module (done [here](https://github.com/flyway/flyway/commit/825fe8909a976960e20b6be436fba50096760223#diff-43cef596c3935b48c4413ed38829e9a9f411a79de7736ea0ee85abab567b0d7e)).

If you add 

```xml
		<dependency>
			<groupId>org.flywaydb</groupId>
			<artifactId>flyway-mysql</artifactId>
		</dependency>
```

to your `pom.xml`, everything works as expected.

**Comment by michael-simons** [comment]
Yikes. Thanks for the heads up, I am sure that will hit in another couple of places, too.

It's a bit sad (and absolutely not Quarkus' fault), but there goes the database agnostic build.

Anyhow, I can confirm, this fixes the issue.

**Comment by geoand** [comment]
> Thanks for the heads up

YW!

> It's a bit sad (and absolutely not Quarkus' fault), but there goes the database agnostic build.

Yeah, it seems a like a weird move...

**Comment by gsmet** [comment]
FWIW, it's documented in the migration guide: https://github.com/quarkusio/quarkus/wiki/Migration-Guide-2.7#flyway .

And they did the same for SQL Server in the previous version (which was included in 2.6).

IIUC, they did it to fix license issues but I don't know the specifics.

**Comment by michael-simons** [comment]
Thanks, @gsmet I should have read this.
